### PR TITLE
Fix all new mypy errors

### PR DESCRIPTION
**Pull request by orsinium**
I've been away for 10 months, and you've already disabled mypy 🙃 

The PR fixes all new mypy errors, to my best effort. In follow-up PRs I'm going to get mypy back on CI and upgrade it to the latest version.

Please, notice what I did with `TcpTransport.connect_tls`. Mypy tells me this code doesn't work, and I agree. Well, I've thrown an assertion for now that I'm pretty sure won't work, but it silents mypy for now. If I understand correctly, this code is not used, not tested, and is just a draft. But I might be wrong.

### CODE REVIEW

**Comment by orsinium** [comment]
Did I break a test? 👀 I'll have a look on Sunday.

File: nats/aio/transport.py
Comment (wallyqs) [inline nats/aio/transport.py:144]: what is the error reported by mypy? This would have been used here: https://github.com/nats-io/nats.py/blob/48059d5191fc8626e6abc85509c54413319c3585/nats/aio/client.py#L1893-L1898

Comment (orsinium) [reply nats/aio/transport.py:144]: `self._io_writer` is defined as `Optional[StreamWriter]` and is defined in 3 places:

1. In `__init__` as None
2. In `connect` using `asyncio.open_connection`.
3. In `connect_tls` at line 169 but AFTER it is used at line 156.

So, either line 156 fails with AttributeError or `self._io_writer` is defined somewhere else. Is it defined? Is `connect` supposed to be called before `connect_tls`?

**Comment by orsinium** [comment]
All tests pass locally for me (except websocket ones, but I guess my nats-server isn't new enough). Is CI flaky?

File: nats/aio/transport.py
Comment (wallyqs) [reply nats/aio/transport.py:144]: I think so, first `connect` would happen in `select_next_server` which would establish the TCP connection, then after getting the initial `INFO` it would upgrade with `connect_tls`:
https://github.com/nats-io/nats.py/blob/48059d5191fc8626e6abc85509c54413319c3585/nats/aio/client.py#L1302-L1308

**Comment by wallyqs** [comment]
yes, some of the `PullSubscribeTest` looks like are flaky...

File: nats/aio/transport.py
Comment (orsinium) [reply nats/aio/transport.py]: Thank you! Fixed: https://github.com/nats-io/nats.py/pull/384/commits/a826001f5a2e13ba34ffc1e158979d232370a1d2

File: nats/aio/client.py
Comment (wallyqs) [inline nats/aio/client.py:130]: typo on handle 😉 
```suggestion
    # TODO(@orsinium ): use cached_property
```

File: nats/aio/transport.py
Comment (wallyqs) [inline nats/aio/transport.py:145]: yes think we can remove the check at this point.

**Review by wallyqs — APPROVED** [review]
LGTM! will merge after format fixes shown on CI, but tests and mypy looks good to me locally 👍 ✅  
```
pipenv run mypy
Success: no issues found in 21 source files
```
Thanks for the contribution again!

File: nats/aio/transport.py
Comment (wallyqs) [inline nats/aio/transport.py:154]: This fails because apparently for uvloop, `uvloop.loop._SSLProtocolTransport` is not an instance of `asyncio.WriteTransport`. I will remove these asserts before merging for now, tests passing locally otherwise.

Comment (wallyqs) [reply nats/aio/transport.py:154]: oh I see but then with mypy start to get the following...
```
nats/aio/transport.py:170: error: Item "None" of "Optional[StreamReader]" has no attribute "read"  [union-attr]
```

File: nats/js/client.py
Comment (wallyqs) [inline nats/js/client.py:293]: this bug would have been caught by mypy...
```
nats/js/client.py:295: error: "ConsumerConfig" has no attribute "memory_storage"; maybe "mem_storage"?  [attr-defined]
```

**Comment by wallyqs** [comment]
Ok merged manually including this to fix the uvloop issue: https://github.com/nats-io/nats.py/commit/c99fdf16de2a9dc712ba36f21f8b4baf788ba2f4
Thanks again!

**Pull request closed without merging**
